postcommit.py

```diff
@@ -0,0 +1,642 @@
+
+
+class GitRepository():
+
+	def __init__(self, local_repository):
+		self.logs = []
+		self.module_dir = local_repository
+        	self.oldId = oldId
+        	self.newId = newId
+        	self.branch = branch
+        	self.module = os.path.basename(self.module_dir)
+        	self.newchangesets = []
+ 
+# 	***************************************************************************************************************
+#	getCommits :
+#
+#			Initial method in this class to get commits,newchangesets json array
+#
+#			@param : 
+#			
+#			@return : {'commits': [{'files': {'Removed': [], 'Added': [{'AddedLines': '1', 'File': 'test',
+#				   'RemovedLines': '0'}], 'Modified': []}, 'changeset': '3e4983962f67', 'parents': 
+#				   'changed_lines': [1], 'file': 'test'}]}], 'user': 'gomathinayagam.d@zohocorp.com', 
+#				   'branch': 'master', 'push_time': '2018-02-28 17:43:45', 'commit_type': 1, 
+#				   'message': 'test', 'commit_date': '2018-02-28 17:43:21', 'revision': '3e49839'}], 
+#				   'newchangesets': ['3e4983962f6743f1fc55ce53e62553a5f83524c8']}
+#       						
+#	*************************************************************************************************************** 
+
+    	def getCommits(self):
+       	 	self.branchlog(os.path.basename(self.branch),self.oldId,self.newId)
+        	return self.logs, self.newchangesets
+
+	
+#       ***************************************************************************************************************
+#	findNewBranchCommits :
+#
+#			Find number of commits present in inital branch creation
+#
+#			@param : changeset
+#			@param : max_count
+#			@param : skip
+# 			@param : newId
+#			@param : branchname
+#			
+#			@return : boolean       						
+#	*************************************************************************************************************** 
+	def findNewBranchCommits(self,changeset,max_count,skip,newId=None,branchname=None):
+		cmd = "git rev-list --first-parent --max-count="+ str(max_count) + "--skip="+ str(skip) +" "+ newId
+		validate = False
+		rev_list = status_check(cmd,self.module_dir)
+		rev_list = str(rev_list[0]).split('\n')
+		rev_list = list(filter(None, rev_list))
+		for x in rev_list:	 					# Iterate until branch base point
+			rev_cmd = "git branch --contains " + x
+			rev_cmd = status_check(rev_cmd,self.module_dir)
+			rev_cmd = str(rev_cmd[0]).split('\n')
+			if(len(rev_cmd) > 2): 
+				validate = True
+				break
+			else:
+				changeset.append(x)
+		return validate
+
+	
+#       ***************************************************************************************************************
+#	branchlog :
+#		
+#			Find a number of commits between tow changsets
+#
+#			@param : branchname
+#			@param : oldId
+#			@param : newId
+#			
+#			@return : None        						
+#	*************************************************************************************************************** 
+    	def branchlog(self, branchname=None, oldId=None,newId=None):
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this function to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS3776)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this parameter "oldId" to match the regular expression ^[_a-z][a-z0-9_]*$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS117)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this parameter "newId" to match the regular expression ^[_a-z][a-z0-9_]*$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS117)